packages/minimongo/minimongo.js

```diff
@@ -696,12 +696,33 @@ LocalCollection.prototype.update = function (selector, mod, options, callback) {
   // it. (We don't need to save the original results of paused queries because
   // they already have a resultsSnapshot and we won't be diffing in
   // _recomputeResults.)
-  var qidToOriginalResults = {};
+  var qidToOriginalResults = {},
+      docMap = {}, // We should only clone each document once, even if it appears in multiple queries
+      onlyIDQuery = _.size(selector) === 1 && _.isString(selector._id); // If the selector is just looking up a single _id, then we only need to clone that document
   _.each(self.queries, function (query, qid) {
     // XXX for now, skip/limit implies ordered observe, so query.results is
     // always an array
-    if ((query.cursor.skip || query.cursor.limit) && ! self.paused)
-      qidToOriginalResults[qid] = EJSON.clone(query.results);
+    if ((query.cursor.skip || query.cursor.limit) && ! self.paused) {
+
+      // This is to catch LocalCollection._IdMap
```

Comment:
Wait, how can this actually ever happen? We're in a block where `query.cursor.skip || query.cursor.limit` is true which implies this is an ordered query (I wish we had more assertions in our codebase). Did you put this clause in defensively or did you actually hit this?